include/exadg/fluid_structure_interaction/driver.cpp

```diff
@@ -281,16 +292,18 @@ Driver<dim, Number>::apply_dirichlet_neumann_scheme(VectorType &       d_tilde,
   fluid->solve_ale();
 
   // update velocity boundary condition for fluid
-  coupling_structure_to_fluid(iteration == 0);
+  coupling_structure_to_fluid(iteration);
 
   // solve fluid problem
-  fluid->time_integrator->advance_one_timestep_partitioned_solve(iteration == 0);
+  fluid->time_integrator->advance_one_timestep_partitioned_solve(iteration ==
+                                                                 0 /* use_extrapolation */);
```

Comment:
I am not trying to implement the idea that we want to use the old solution values at `t_n` as starting guess for the fluid and structure solvers. This we might need for future comparison, but the present methodology to use the last iterates in the individual solvers makes a lot of sense (one of the points we wanted to address in the planned FSI paper).

What I am aiming for in this PR is:

keep the logic for re-using iterates in the individual field solvers, 
and
just for the first FSI coupling loop, 
make the use of an extrapolated structure velocity at the interface optional, since it is the first thing used to couple (structure -> ALE and structure -> fluid)
for the ALE, this was missing (as this is never really the problem, but the incompressible fluid might "overreact", which is why this PR might be interesting for larger/varying time steps)

Reply:
I think I understood it correctly, at least the basic intention. The only thing I am not sure about is that you are mentioning that only the structure velocity is affected by this PR. According to my understanding, also the displacements and `coupling_structure_to_ale(d);` are affected by this PR, because

1. call `update_structure_displacement(d, k);` -> calculation of displacement affected by the present PR
2. call `apply_dirichlet_neumann_scheme(d_tilde, d, k);` given `d` from 1., which then calls immediately
3. `coupling_structure_to_ale(d);` -> hence also affected by the present PR ?!
4. having solved the ALE problem, call `coupling_structure_to_fluid(iteration);`, where the calculation of the structure velocity is affected by the present PR

Do you agree?

Reply:
yes this is all correct, it should be "make use of an extrapolated structure velocity _and displacement_ at the interface optional". I have corrected the comment (marked bold).
The reason why I did not wrap the ALE update like:
`if(iteration > 0 or this->parameters.use_extrapolation)`
`{`
`  coupling_structure_to_ale(displacement_structure);`
`  fluid->solve_ale();`
`  coupling_structure_to_fluid(iteration);`
 `}`
is that the previous iteration does not update the fluid mesh anymore after solving the structure in the Dirichlet-Neumann coupling loop and reaching convergence. Hence, also if no extrapolation is used, that is, using the structural displacement at the previous time `t_n`, which is the last recent iterate (which lead to convergence), the mesh has to be updated. So all cases require an ALE update in my opinion.

Reply:
the `bool use_extrapolation` was indeed misleading I think.
One could not know, if we mean the FSI coupling loop initial guess or the individual field solvers!
the enum name is now quite long, but I use autocomplete anyways, hence prefer a more descriptive name.
but what do you think about:

`enum class InitialGuessFSI`
`{`
`  ExtrapolatedInTime,`
`  LastTimestepValue`
`}`

and hence a variable `initial_guess_fsi` (I always derive the variable name from the enum name to find it easily)